app/scripts/views/sign_up.js

```diff
@@ -51,6 +51,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, Session, AuthErrors,
         return p(false);
       }
 
+      /**
+       * check if there is an account we can suggest for an OAuth flow.
+       */
+      var account = this.user.getChooserAccount();
+      var serviceName = this.relier.get('serviceName');
+
+      if (account.get('email') && serviceName) {
```

Comment:
> we have an endpoint that makes this decision already

Does this already exist and work?  The commentary in #2101 seems to suggest that it's a WIP.